src/fluids.md

```diff
@@ -0,0 +1,184 @@
+# Navier-Stokes Mini Application
+
+The solver implemented in this miniapp solves the transient incompressible
+Navier-Stokes equations.
+
+## Theory
+
+The equations are given in the non-dimensionalized form
+
+\begin{align}
+    \frac{\partial u}{\partial t} + (u \cdot \nabla) u - \frac{1}{Re} \nabla^2 u + \nabla p &= f & \quad \text{in } \Omega\\\\
+    \nabla \cdot u &= 0 & \quad \text{in } \Omega
+\end{align}
+
+where $Re$ represents the Reynolds number. In order to solve these equations,
+the method presented in Tomboulides (1997)[^1] is used, which is based on an
+equal order finite element discretization on quadrilateral or hexahedral
+elements of high polynomial order. The method describes an implicit-explicit
+time-integration scheme for the viscous and convective terms respectively.
+Introducing the following notation for the linear term $L(u) = \frac{1}{Re}
+\nabla u$ and the nonlinear term $N(u) = -(u \cdot \nabla) u$ and their
+time-extrapolated forms
+
+\begin{align}
+    L^\*(u^{n+1}) = \sum_{j=1}^k a_j L(u^{n+1-j}) \\\\
+    N^\*(u^{n+1}) = \sum_{j=1}^k a_j N(u^{n+1-j})
+\end{align}
+
+where $a_j$ are coefficients from the corresponding explicit time integration
+method.
+
+Applying a BDF method to the initial equation using the introduced forms yields
+
+\begin{align}
+    \sum_{j=0}^k \frac{b_j}{\Delta t} u^{n+1-j} =
+    -\nabla p^{n+1} + L(u^{n+1}) + N^\*(u^{n+1}) + f^{n+1}.
+\end{align}
+
+Collecting all known quantities at a given time with
+
+\begin{align}
+    F^\*(u^n) = -\sum_{j=1}^k \frac{b_j}{\Delta t} u^{n+1-j}
+    + N^\*(u^{n+1}) + f^{n+1}
+\end{align}
+
+the BDF expression reduces to
+
+\begin{align}
+    \label{eq:bdf_short}
+    \frac{b_0}{\Delta t} u^{n+1} = -\nabla p^{n+1} + L(u^{n+1}) + F^\*(u^n).
+\end{align}
+
+To achieve a high order convergence in space the linear term $L(u)$ is replaced
+by
+
+\begin{align}
+    L_{\times}(u) = \nu \nabla(\nabla \cdot u) - \nu \nabla \times \nabla \times u
+\end{align}
+
+which is used to weakly enforce incompressibility by setting the first term to
+zero.
+
+To compute the pressure we rearrange \eqref{eq:bdf_short} and take the
+divergence on both sides
+
+\begin{align}
+    \label{eq:prespois}
+    \nabla^2 p^{n+1} = \nabla \cdot (L_{\times}^\*(u^{n+1} + F^\*(u^{n+1})),
+\end{align}
+
+which is closed by the Neumann type boundary condition
+
+\begin{align}
+    \nabla^2 p^{n+1} \cdot \hat{n} = -\frac{b_0}{\Delta t} \cdot \hat{n}
```

Comment:
It looks like the terms dotted with \hat{n} are scalars.  Perhaps the divergence is not need in these terms?

Reply:
True, that's a copy paste error.